eval/tricks.rkt

```diff
@@ -8,54 +8,57 @@
          "../ops/all.rkt"
          "machine.rkt")
 
-(provide get-ampls
+(provide get-bounds
          get-slack)
 
-(define (get-slack)
-  (match (*sampling-iteration*)
-    [1 512]
-    [2 1024]
-    [3 2048]
-    [4 4096]
-    [5 8192]))
+(define (get-slack [iter (*sampling-iteration*)])
+  (match iter
+    [0 0]
+    [_ (* (expt 2 (- iter 1)) 512)]))
 
 (define (crosses-zero? x)
   (not (equal? (mpfr-sign (ival-lo x)) (mpfr-sign (ival-hi x)))))
 
-; We assume the interval x is valid. Critical not to take mpfr-exp of inf or 0,
-; the results are platform-dependant
-(define (maxlog x)
+; We assume the interval x is valid. Critical not to take mpfr-exp of inf or 0
+(define (maxlog x #:less-slack [less-slack #f])
+  (define iter
+    (if less-slack
+        (- (*sampling-iteration*) 1)
+        (*sampling-iteration*)))
```

Comment:
Why?

Reply:
I thought it this way:

We have uncertainties in Rival, and all of them are known after the first pass. 
It is always the case (I assume), we can get rid of uncertainties but we can not get new ones.

Therefore,
What to do with lower bound when we have these uncertainties, just do not consider slack at all when calculating lower bound?
Or what lower bound should be for the formulas that we do not tune?

Lower bound for the formulas that we do not tune should be:
"we have tried `512` bits of slack last iter, now try `1024` and use `512` as a lower bound since we have not converged with `512`".

For the _uncertainties_ I also thought quite similar:
"last iteration we used slack `512` for `(cos ...) = [-1, 1]` and we have still have this unsolved uncertainty. well, now increase slack to `1024` but treat `512` as an addition to _lower bound_, since we have tried it and it didn't work".
I feel sketchy about this one, but this is how I implemented it.

Reply:
Got it, this actually makes sense and is smart. I'm not thrilled about the function signature here but let's merge it for now.